tests/queries/0_stateless/02346_into_outfile_and_stdout.sh

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+
+function perform()
+{
+    local test_id=$1
+    local query=$2
+
+    echo "performing test: ${test_id}"
+    ${CLICKHOUSE_CLIENT} --query "${query}"
+    if [ "$?" -eq 0 ]; then
+        cat "${CLICKHOUSE_TMP}/test_into_outfile_and_stdout_${test_id}.out"
+    else
+        echo "query failed"
+    fi
+    rm -f "${CLICKHOUSE_TMP}/test_into_outfile_and_stdout_${test_id}.out"
+}
+
+function performBadQuery()
+{
+    local test_id=$1
+    local query=$2
+    local error_message=$3
+
+    echo "performing test: ${test_id}"
+    ${CLICKHOUSE_CLIENT} --query "${query}" 2>&1 |  grep -Fc "${error_message}"
+}
+
+function performFileExists()
+{
+    local test_id=$1
+    local query=$2
+    local error_message=$3
+
+    touch "${CLICKHOUSE_TMP}/test_into_outfile_and_stdout_${test_id}.out"
+
+    echo "performing test: ${test_id}"
+    ${CLICKHOUSE_CLIENT} --query "${query}" 2>&1 |  grep -Fc "${error_message}"
+    rm -f "${CLICKHOUSE_TMP}/test_into_outfile_and_stdout_${test_id}.out"
+}
+
+function performCompression()
+{
+    local test_id=$1
+    local query=$2
+
+    echo "performing test: ${test_id}"
+    ${CLICKHOUSE_CLIENT} --query "${query}"
+    if [ "$?" -eq 0 ]; then
+        gunzip "${CLICKHOUSE_TMP}/test_into_outfile_and_stdout_${test_id}.gz"
+        cat "${CLICKHOUSE_TMP}/test_into_outfile_and_stdout_${test_id}"
+    else
+        echo "query failed"
+    fi
+    rm -f "${CLICKHOUSE_TMP}/test_into_outfile_and_stdout_${test_id}"
+}
+
+
+
+perform "select" "SELECT 1, 2, 3 INTO OUTFILE '${CLICKHOUSE_TMP}/test_into_outfile_and_stdout_select.out' AND STDOUT"
```

Comment:
Let's also write a test for the case when we insert more than one buffer size, e.g. `nextImpl()` should be called more than once.

Reply:
Based on my understanding of your comment I have added union test case which calls nextImpl() twice. Hope this addresses it.